### PR TITLE
[skrifa] autohint shaper abstraction

### PR DESCRIPTION
**Pull request by dfrg**
Adds the infrastructure to support shaping without making any functional changes.

A little bit of churn but the intention is to allow future PRs to be more focused.

### CODE REVIEW

**Review by drott — APPROVED** [review]
LGTM - do we later still want Nominal? Is this transitional?

**Comment by dfrg** [comment]
Nominal allows us to keep testing against a known FreeType configuration which will at least ensure that the internal algorithms are a match. It's unlikely that the "best effort" mode will ever truly match FT+HB so I think we should keep nominal around until some future point that we have access to a full shaper.

File: skrifa/src/outline/autohint/shape.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+pub(crate) struct ShapedGlyph {
+    pub id: GlyphId,
+    pub y_offset: i32,
+}
+
+/// Arbitrarily chosen to cover our max input size plus some extra to account
+/// for expansion from multiple substitution tables.
+const SHAPED_CLUSTER_INLINE_SIZE: usize = 16;
+
+pub(crate) type ShapedCluster = SmallVec<ShapedGlyph, SHAPED_CLUSTER_INLINE_SIZE>;
```
Comment (rsheeter) [inline skrifa/src/outline/autohint/shape.rs:32]: cluster is not so widely known outside of a very small circle, suggest adding a reference that explains what clusters are

File: skrifa/src/outline/autohint/shape.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
+//! Shaping support for autohinting.
+
+use raw::{tables::gsub::Gsub, TableProvider};
+
+use crate::{charmap::Charmap, collections::SmallVec, FontRef, GlyphId, MetadataProvider};
+
+/// Determines the fidelity with which we apply shaping in the
+/// autohinter.
```
Comment (rsheeter) [inline skrifa/src/outline/autohint/shape.rs:8]: I think I'd expect an enum that determines fidelity to give clearer guidance about the tradeoffs and/or which is better, can we beef up the comment a touch?

**Review by rsheeter — APPROVED** [review]

**Pull request merged**